sshuttle/methods/pf.py

```diff
@@ -108,16 +111,21 @@ def pf_query_nat(family, proto, src_ip, src_port, dst_ip, dst_port):
     pnl.direction = PF_OUT
     pnl.af = family
     memmove(addressof(pnl.saddr), packed_src_ip, length)
-    pnl.sxport.port = socket.htons(src_port)
     memmove(addressof(pnl.daddr), packed_dst_ip, length)
-    pnl.dxport.port = socket.htons(dst_port)
+    if sys.platform == 'darwin':
+        pnl.sxport.port = socket.htons(src_port)
+        pnl.dxport.port = socket.htons(dst_port)
+    else:
+        pnl.sxport = socket.htons(src_port)
+        pnl.dxport = socket.htons(dst_port)
```

Comment:
In FreeBSD the fields sport, dport, rsport, rdport are just u_int16_t, not a union as in OSX.